sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -399,6 +395,52 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
   }
 
   /**
+   * Loads an `Dataset[String]` storing CSV rows and returns the result as a `DataFrame`.
+   *
+   * Unless the schema is specified using `schema` function, this function goes through the
+   * input once to determine the input schema.
+   *
+   * @param csvDataset input Dataset with one CSV row per record
+   * @since 2.2.0
+   */
+  def csv(csvDataset: Dataset[String]): DataFrame = {
+    val parsedOptions: CSVOptions = new CSVOptions(
+      extraOptions.toMap,
+      sparkSession.sessionState.conf.sessionLocalTimeZone)
+    val filteredLines = CSVUtils.filterCommentAndEmpty(csvDataset, parsedOptions)
+    val maybeFirstLine = filteredLines.take(1).headOption
+    if (maybeFirstLine.isEmpty) {
+      return sparkSession.emptyDataFrame
```

Comment:
The reason why this one exists unlike json is, CSV needs to head a head always first (even if it does not infer the schema, it needs at least the number of values). In this case, we could return empty one fast.